pkg/runtime.v2/framework/plugins/jobset/jobset.go

```diff
@@ -126,6 +127,24 @@ func jobSetIsSuspended(jobSet *jobsetv1alpha2.JobSet) bool {
 	return ptr.Deref(jobSet.Spec.Suspend, false)
 }
 
+func (j *JobSet) TerminalCondition(ctx context.Context, trainJob *kubeflowv2.TrainJob) (*metav1.Condition, error) {
```

Comment:
As we discussed here: https://github.com/kubeflow/training-operator/pull/2298#discussion_r1826259033, does JobSet's Failed or Completed condition represents TerminalPhase ?
E.g. we don't need to check the `.status.terminalState` ?